general/arrays_by_name.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "arrays_by_name.hpp"
+
+namespace mfem
+{
+
+template <class T>
+void ArraysByName<T>::Print(std::ostream &os, int width) const
```

Comment:
Is there a reason these are in the cpp file and only instantiated for `ArraysByName<T>` (rather than having everything in the hpp file)?

Reply:
The main reason is that I struggled with supporting different printed  formats which would add complexity to the Print method and I didn't want these longer methods in the header (I also wanted the files to be recompiled while I was testing formats without manually deleting select object files). Now that the formats are more or less settled I can move them into the header. I can also add more instantiation types.

Reply:
No need to add more (or any, really) instantiations — if it's in the header, the templates will be instantiated automatically when they're used. (And without `extern` declarations, I don't think it will speed up compile times).